### ISSUE TITLE
Future Request: save selected image to file

### ISSUE THREAD

**Issue by yurix4** [sumatrapdfreader/sumatrapdf#542]
Good for CBR/CBZ files.

**Comment by SumatraPeter** [comment]
You mean, extract a particular image? Can't you do that with an archiver (such as 7-Zip or WinRAR)?

**Comment by yurix4** [comment]
It is very uncomfortable. When I see what kind of image you want to save me, I need operdelit filename (Sumatra today does not show the names of the images in the archive), open archiver find this image and extract. Very inconvenient if I sort thousands of images from the archives.

**Comment by SumatraPeter** [comment]
What do you think about the feature request I just made now? (See link above.)

**Comment by yurix4** [comment]
I think that it is a necessary function. Also it would be nice if the map could be not only in the form of text (https://github.com/sumatrapdfreader/sumatrapdf/issues/540).

**Comment by GitHubRulesOK** [comment]
This is best done in a couple of clicks using an ExternalViewer application so here I press ALT F I and that page or image opens in IrfanView

![image](https://user-images.githubusercontent.com/12754913/73580667-4ad0fe00-447e-11ea-837c-defacb396351.png)

Many applications can have different command calls, but this is how I use IrfanView Portable with its PDF plugin for a PDF or single image files
It would need a very different scripted approach to pick out the numbered page from a .zip or .cbx but it should not be impossible given that SumatraPDF numbers pages in natural alpha order.

```
ExternalViewers [
	[
		CommandLine = "C:\Apps\Viewers\IrfanView\irfanview454\x64\i_view64.exe" "%1" /page=%p
		Name = &IrfanView Image editor
		Filter = *.pdf;*.gif;*.png;*.jpg;*.tga;*.tif;*.webp
	]
]
```

**Comment by SumatraPeter** [comment]
Perhaps when #1336 is implemented then as an add-on this can be done too... Now that #543 is done a Save As option could be added to the context menu of each image listed as a bookmark entry.

**Comment by GitHubRulesOK** [comment]
@SumatraPeter 
I agree that object extraction / saving is a highly desirable function and if possible there should be a way to pass them to other tools, at present the .bmk listing is not extended to all formats (my beef) and image object pointers are understandably not 100% consistent across varied structures of archive,  folder or document embodiment. However your requests have helped since it is possible to parse some of the .bmk files that are currently available.
Which reminds me I need to raise another open issue :-)

**Comment by GitHubRulesOK** [comment]
I have published my SumatraPDF Addin script that exports different file **whole page** types as fixed quality PNG (rather than as inferior lossy JPG) but once extracted, either as a single page or a range of pages, they can easily be converted by any other imaging tool.
For more details see
 https://forum.sumatrapdfreader.org/t/export-save-page-s-as-jpg-png-picture-s-image-s/1030/11

**Comment by GitHubRulesOK** [comment]
This is now possible in 3.7 Pre-release  (too many ways to describe all abilities but )    any image such as PDF HEIC WebP or simpler CBx can be saved cropped or resized such as this High Definition TIFF saved as a trimmed down PNG using Arrow keys for exact size and positioning

<img width="1900" height="750" alt="Image" src="https://github.com/user-attachments/assets/867b1abc-e116-40d7-83bd-5a25e5965c4f" />
Even as JPEG it will be 90% high Quality This even if you open in another tab  is just half the size. so is 1600 x 1200 saved as full size (cropped from 2400x1335).

<img width="809" height="632" alt="Image" src="https://github.com/user-attachments/assets/19b2f578-cb0f-4d70-8331-63b85e1e4818" />
